src/utils/pattern.ts

```diff
@@ -9,7 +9,7 @@ const GLOBSTAR = '**';
 const ESCAPE_SYMBOL = '\\';
 
 const COMMON_GLOB_SYMBOLS_RE = /[*?]|^!/;
-const REGEX_CHARACTER_CLASS_SYMBOLS_RE = /\[.*]/;
+const REGEX_CHARACTER_CLASS_SYMBOLS_RE = /\[[^[]*]/;
```

Comment:
One thing to test is to make sure this handles `[` inside of brackets as expected. If provided `'[[]'`, the current regex matches `'[[]'` (the whole string), but the regex proposed here only matches the last two characters (`'[]'`). I don't know if one or the other would be considered a bug, or if `'[[]'` is malformed and the behavior there is undefined.

Reply:
The main goal of this regex is to say that the pattern contains a character classes (`[1-5]`, `[:alnum:]`, …). Based on this statement, I think it is enough here to check that there is something enclosed in square brackets in the pattern. We don't need to check that the pattern is correct (syntax and nesting checking). Pattern validation is a separate layer delegated to the `micromatch`  package. In this case, your proposed solution is correct.